File: component/biz/Class_biz_bureaus.pas
unit Class_biz_bureaus;

interface

uses
  Class_db_bureaus,
  kix;

type
  TClass_biz_bureaus = class
  published
    function Bind
      (
      partial_code: string;
      target: system.object
      )
      : boolean;
    procedure BindDirectToListControl
      (
      target: system.object;
      unselected_literal: string = '-- bureau --';
      selected_value: string = EMPTY
      );
    function Delete(code: string): boolean;
    function Get
      (
      code: string;
      out description: string
      )
      : boolean;
    procedure &Set
      (
      code: string;
      description: string
      );
  public
    constructor Create;
  strict private
    db_bureaus: TClass_db_bureaus;
  end;

implementation

constructor TClass_biz_bureaus.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
  db_bureaus := TClass_db_bureaus.Create;
end;

function TClass_biz_bureaus.Bind
  (
  partial_code: string;
  target: system.object
  )
  : boolean;
begin
  Bind := db_bureaus.Bind(partial_code,target);
end;

procedure TClass_biz_bureaus.BindDirectToListControl
  (
  target: system.object;
  unselected_literal: string = '-- bureau --';
  selected_value: string = EMPTY
  );
begin
  db_bureaus.BindDirectToListControl(target,unselected_literal,selected_value);
end;

function TClass_biz_bureaus.Delete(code: string): boolean;
begin
  Delete := db_bureaus.Delete(code);
end;

function TClass_biz_bureaus.Get
  (
      code: string;
      out description: string
  )
  : boolean;
begin
  //
  Get := db_bureaus.Get
    (
    code,
    description
    );
  //
end;

procedure TClass_biz_bureaus.&Set
  (
      code: string;
      description: string
  );
begin
  //
  db_bureaus.&Set
    (
    code,
    description
    );
  //
end;

end.

File: component/biz/Class_biz_bureaus.cs
using Class_db_bureaus;
using kix;

namespace Class_biz_bureaus
  {
  public class TClass_biz_bureaus
    {

        private readonly TClass_db_bureaus db_bureaus = null;

        //Constructor  Create()
        public TClass_biz_bureaus() : base()
        {
            // TODO: Add any constructor code here
            db_bureaus = new TClass_db_bureaus();
        }
        public bool Bind(string partial_code, object target)
        {
            bool result;
            result = db_bureaus.Bind(partial_code, target);
            return result;
        }

        public void BindDirectToListControl(object target, string unselected_literal, string selected_value)
        {
            db_bureaus.BindDirectToListControl(target, unselected_literal, selected_value);
        }

        public void BindDirectToListControl(object target)
        {
            BindDirectToListControl(target, "-- Bureau --");
        }

        public void BindDirectToListControl(object target, string unselected_literal)
        {
            BindDirectToListControl(target, unselected_literal, k.EMPTY);
        }

        public bool Delete(string code)
        {
            bool result;
            result = db_bureaus.Delete(code);
            return result;
        }

        public bool Get(string code, out string description)
        {
            bool result;
            result = db_bureaus.Get(code, out description);

            return result;
        }

        public void Set(string code, string description)
        {
            db_bureaus.Set(code, description);

        }

    } // end TClass_biz_bureaus

}
